src/main/java/seedu/uninurse/logic/parser/FindCommandParser.java

```diff
@@ -25,9 +34,30 @@ public FindCommand parse(String args) throws ParseException {
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
         }
 
-        String[] nameKeywords = trimmedArgs.split("\\s+");
-
-        return new FindCommand(new NameContainsKeywordsPredicate(Arrays.asList(nameKeywords)));
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS,
+                        PREFIX_CONDITION, PREFIX_TASK_DESCRIPTION, PREFIX_TAG);
+
+        String[] keywords = argMultimap.getPreamble().trim().split("\\s+");
+
+        return new FindCommand(new PatientMatchPredicate(
+                sanitizeList(Arrays.asList(keywords)),
+                sanitizeList(argMultimap.getAllValues(PREFIX_NAME)),
+                sanitizeList(argMultimap.getAllValues(PREFIX_PHONE)),
+                sanitizeList(argMultimap.getAllValues(PREFIX_EMAIL)),
+                sanitizeList(argMultimap.getAllValues(PREFIX_ADDRESS)),
+                sanitizeList(argMultimap.getAllValues(PREFIX_TAG)),
+                sanitizeList(argMultimap.getAllValues(PREFIX_CONDITION)),
+                sanitizeList(argMultimap.getAllValues(PREFIX_TASK_DESCRIPTION))));
     }
 
+    /**
+     * Adds an empty string if the list is empty.
+     */
+    private List<String> sanitizeList(List<String> list) {
+        if (list.isEmpty()) {
+            list.add("");
+        }
+        return list;
+    }
```

Comment:
May I ask what is the purpose of this method because it looks kind of redundant. Would it not suffice to just pass an empty list into the predicates?

Reply:
Originally, predicate returns false if keyword is empty.